src/main/java/com/hms_networks/americas/sc/extensions/system/threading/SCCountdownLatch.java

```diff
@@ -0,0 +1,43 @@
+package com.hms_networks.americas.sc.extensions.system.threading;
+
+/**
+ * Class implementation of a CountDownLatch for Java 1.4.
+ *
+ * @since 1.4.0
+ * @author HMS Networks, MU Americas Solution Center
+ */
+public class SCCountdownLatch {
+
+  /** The count of the latch. */
+  private int count;
+
+  /**
+   * Constructs a new {@code SCCountdownLatch} initialized with the given count.
+   *
+   * @param count the number of times {@link #countDown} must be invoked before threads can pass
+   *     through {@link #await}.
+   * @throws IllegalArgumentException if {@code count} is negative.
+   */
+  public SCCountdownLatch(int count) {
+    this.count = count;
+  }
+
+  /**
+   * Waits for the latch to count down to zero, unless the thread is interrupted.
+   *
+   * @throws InterruptedException if the current thread is interrupted
+   */
+  public synchronized void await() throws InterruptedException {
```

Comment:
Consider overload await(long timeout).  This would allow callers to pass a timeout.

Reply:
added imeout overload